collector/sbd/sbd.go

```diff
@@ -132,3 +146,57 @@ func (c *sbdCollector) getSbdDeviceStatuses(sbdDevices []string) map[string]stri
 
 	return sbdStatuses
 }
+
+// for each sbd device, extract the watchdog timeout via regex
+func (c *sbdCollector) getSbdWatchDogTimeout(sbdDevices []string) map[string]float64 {
+	sbdWatchdogs := make(map[string]float64)
+	for _, sbdDev := range sbdDevices {
+		sbdDump, _ := exec.Command(c.sbdPath, "-d", sbdDev, "dump").Output()
+
+		regex := regexp.MustCompile(`Timeout \(watchdog\)  *: \d+`)
+		// we get this line:		Timeout (watchdog) : 5
+		watchdogLine := regex.FindStringSubmatch(string(sbdDump))
+
+		if watchdogLine == nil {
+			continue
+		}
+		// get the timeout from the line
+		regexNumber := regexp.MustCompile(`\d+`)
+		watchdogTimeout := regexNumber.FindStringSubmatch(string(watchdogLine[0]))
+		if watchdogTimeout == nil {
+			continue
+		}
+		// map the timeout to the device
+		if s, err := strconv.ParseFloat(watchdogTimeout[0], 64); err == nil {
+			sbdWatchdogs[sbdDev] = s
+		}
+	}
+	return sbdWatchdogs
+}
+
+// for each sbd device, extract the msgWait timeout via regex
+func (c *sbdCollector) getSbdMsgWaitTimeout(sbdDevices []string) map[string]float64 {
+	sbdMsgWaits := make(map[string]float64)
+	for _, sbdDev := range sbdDevices {
+		sbdDump, _ := exec.Command(c.sbdPath, "-d", sbdDev, "dump").Output()
+
+		regex := regexp.MustCompile(`Timeout \(msgwait\)  *: \d+`)
+		// we get this line:		Timeout (msgwait) : 5
+		msgWaitLine := regex.FindStringSubmatch(string(sbdDump))
+
+		if msgWaitLine == nil {
+			continue
+		}
+		// get the timeout from the line
+		regexNumber := regexp.MustCompile(`\d+`)
+		msgWaitTimeout := regexNumber.FindStringSubmatch(string(msgWaitLine[0]))
+		if msgWaitTimeout == nil {
+			continue
+		}
+		// map the timeout to the device
+		if s, err := strconv.ParseFloat(msgWaitTimeout[0], 64); err == nil {
+			sbdMsgWaits[sbdDev] = s
+		}
+	}
+	return sbdMsgWaits
+}
```

Comment:
There is a lot of repetition here, maybe we could refactor these two functions into a single one that gets all the timeouts.
The returned map would be something like `map[string]map[string]float64`, where the first key is the device and the second is the type of the timeout.

Reply:
yes for a 1st version I wanted to have something done, and have a high-level pov before solving the details. I will refactor accordingly now

Reply:
yes for a 1st version I wanted to have something done, and have a high-level pov before solving the details. I will refactor accordingly now